ARCHITECTURE.md

```diff
@@ -0,0 +1,52 @@
+## CHAPTER 1
+
+> **Functional Requirements**
+> 
+> Functional requirements describe the system’s behavior: what it does,
+> the user interactions, and the business logic.
+> 
+> Define the problem, why solving it is relevant, and any vital context.
+> Make a list of questions about the business rules you want answered.
+> 
+> If you have a long list of functional requirements, split them by
+> context (e.g., consumer/logistics or deployment/scaling). Solutions
+> will usually follow these splits, so you may delegate the
+> implementation of context’s features to different teams or just place
+> them in different services.
+
+**1. Função principal - Descoberta de arquivos e diretórios**
+
+-   A aplicação precisa pegar uma URL fornecida e testar possíveis arquivos e diretórios nela.
+-   Se encontrar uma página do tipo “Index Of”, em que já esteja listado os arquivos e diretórios, não é necessário continuar com as tentativas (a maioria das ferramentas não tem essa opção e eu senti falta disso, é útil para laboratórios). Eu gostaria que isso fosse até um padrão, mas com opção de poder desligar caso se encontre falsos positivos.
+-   Ao encontrar um diretório, gostaria de ter a opção da aplicação começar a testar o brute force nele também, para encontrar subdiretórios e arquivos.
+
+**2. Sobre output, interface e controles do usuário**
+
+-   A cada arquivo ou diretório que encontrar, exibir na saída o status, tamanho, link completo pra visualização ou que fique prático pra copiar e colar em um navegador
```

Comment:
What if there are too many files/directories? Couldn't the output overflow the terminal buffer, making the user lose the content?
Printing to stdout is a reasonable default, but you should support more scalable output options.